prusti-viper/src/encoder/mir/arrays/encoder.rs

```diff
@@ -0,0 +1,65 @@
+use super::interface::{EncodedArrayTypes, EncodedSliceTypes};
+use crate::encoder::{
+    errors::{EncodingError, EncodingResult},
+    high::types::HighTypeEncoderInterface,
+    Encoder,
+};
+use rustc_middle::ty;
+
+/// Encode types, type predicates and builtin lookup functions required for handling arrays of
+/// type `array_ty`.
+pub fn encode_array_types<'p, 'v: 'p, 'tcx: 'v>(
```

Comment:
```suggestion
pub(crate) fn encode_array_types<'p, 'v: 'p, 'tcx: 'v>(
```
(Also, in other places.)